db/compaction/compaction_job.cc

```diff
@@ -396,29 +523,34 @@ void CompactionJob::GenSubcompactionBoundaries() {
           *(c->mutable_cf_options()), out_lvl,
           c->immutable_options()->compaction_style, base_level,
           c->immutable_options()->level_compaction_dynamic_level_bytes)));
-  uint64_t subcompactions =
-      std::min({static_cast<uint64_t>(ranges.size()),
-                static_cast<uint64_t>(c->max_subcompactions()),
-                max_output_files});
 
-  if (subcompactions > 1) {
-    double mean = sum * 1.0 / subcompactions;
+  // Get the number of planned subcompactions, may update reserve threads
+  // and update extra_num_subcompaction_threads_reserved_
+  uint64_t num_planned_subcompactions = GetNumPlannedSubcompactions(
+      max_output_files, static_cast<uint64_t>(ranges.size()));
+
+  if (num_planned_subcompactions > 1) {
```

Comment:
I wonder if it worths to write this as `if (num_planned_subcompactions <= 1) {return; }` to reduce nested if-else as we are adding more of them.

Reply:
I think this should be ok. Will apply this change.

Reply:
Should've copy/pasted to get the curly braces per style guide. Anyways it's minor.